### PR TITLE
Testingbranch

### PR DESCRIPTION
**Pull request by silkyh13**

### CODE REVIEW

File: client/components/app.jsx [hunk 1]
```diff
@@ -2,94 +2,100 @@ import React from 'react';
...
   render () {
     return (
+      <div>
+        <h3 style={style.header}>{this.state.itemName}</h3>
+        <div className="carousel" style={style.carousel}>
+          <div className="carouselWrapper"
+            style={{display: 'inline-flex',
+              transform: `translateX(${this.state.translateVal}px)`,
+              transition: 'transform ease-out 0.45s'
+            }}>
```
Comment (mflinn92) [inline client/components/app.jsx]: Maybe extract this to a css module

File: client/components/carousel.jsx [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
-const CarouselItem = (props) => {
-
+const CarouselPic = (props) => {
   return (
 
-    <div className="carouselItem" style={{
+    <div className="carouselPic" id="carouselPic" style={{
       backgroundImage: `url(${props.image})`,
       backgroundSize: 'cover',
       backgroundRepeat: 'no-repeat',
```
Comment (mflinn92) [inline client/components/carousel.jsx]: Same here, import this from a css module

File: server/index.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ const port = 3000;
 const app = express();
 app.use(parser.json());
 
-app.use(express.static(__dirname + '/../dist'))
+app.use(express.static(__dirname + '/../dist'));
 //get the names and pics of the products
 app.get('/products', function (req, res) {
```
Comment (mflinn92) [inline server/index.js]: Do you need this route? Will you ever need to retrieve all your products at once?

File: server/models/index.js [hunk 4]
```diff
@@ -33,42 +33,33 @@ let saveProduct = (productId, productItem, pictureUrl) => {
...
   const instance = new MyWishlistModel({
     products: products,
     username: username
   });
-  instance.save((err) => {
-    if (!err) {
-      console.log('success');
-    }
-  });
+  return await instance.save();
```
Comment (mflinn92) [inline server/models/index.js]: When you use async await normally you would want to do somthing like
`try {
  await action()
catch (err) {
  doSomething(err)
}`
Other wise if your async action fails you will get errors that may crash your server

File: server/models/index.js [hunk 5]
```diff
@@ -33,42 +33,33 @@ let saveProduct = (productId, productItem, pictureUrl) => {
...
+let getProducts = async (callback) => {
+  return await MyProductsModel.find({}).sort([['productId', 'ascending']]).exec();
 };
-let getWishlists = (callback) => {
-  MyWishlistModel.find({}, (err, docs) => {
-    callback(err, docs);
-  });
+
+let getWishlists = async (callback) => {
+  return await MyWishlistModel.find({});
```
Comment (mflinn92) [inline server/models/index.js]: If you're going to use async await then you don't need to pass in callbacks

File: testing/client/components/App.test.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import React from 'react';
+import Enzyme from 'enzyme';
+import { shallow } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+Enzyme.configure({adapter: new Adapter()});
+import App from '../../../client/components/app.jsx';
+import Carousel from '../../../client/components/carousel.jsx';
+import Display from '../../../client/components/display.jsx';
+import LeftArrow from '../../../client/components/leftArrow.jsx';
+import RightArrow from '../../../client/components/rightArrow.jsx';
```
Comment (mflinn92) [inline testing/client/components/App.test.js]: Do you plan on making test for these at some point?

File: testing/sum.test.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+const sum = require('../testjs.js');
+
```
Comment (mflinn92) [inline testing/sum.test.js]: Maybe get rid of this now

File: server/models/index.js (hunk 4 above)
Comment (mflinn92) [reply server/models/index.js]: I don't know why line breaks didn't show up in this. 
try {
  await action();
} catch (err) {
  doSomething(err)
}

Comment (silkyh13) [reply server/models/index.js]: I am not sure about this but I will keep it in mind.

**Pull request merged**